### PR TITLE
Merge Features/compact serialization

### PR DESCRIPTION
**Pull request by yuce**
Merges the features/compact branch into `main`.

### CODE REVIEW

File: internal/serialization/field_operations.go [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+type ArrayOfNullableFloat64FieldKindOperation struct{}
+
+func (ArrayOfNullableFloat64FieldKindOperation) KindSizeInBytes() int32 {
+	return variableKindSize
+}
+
+func FieldOperations(fieldKind pserialization.FieldKind) FieldKindOperation {
+	switch fieldKind {
+	case pserialization.FieldKindBoolean:
+		return BooleanFieldKindOperation{}
```
Comment (mdumandag) [inline internal/serialization/field_operations.go:287]: minor: maybe we can put these into an array for much faster lookups

File: serialization/serialization_config.go [hunk 2]
```diff
@@ -26,6 +26,7 @@ import (
 type Config struct {
 	globalSerializer                    Serializer
 	customSerializers                   map[reflect.Type]Serializer
+	Compact                             CompactConfig
```
Comment (mdumandag) [inline serialization/serialization_config.go:29]: why is this public, while the others are private?

**Review by mdumandag — APPROVED** [review]

File: internal/serialization/field_operations.go (hunk 1 above)
Comment (yuce) [reply internal/serialization/field_operations.go:287]: That makes sense.

File: serialization/serialization_config.go (hunk 2 above)
Comment (yuce) [reply serialization/serialization_config.go:29]: We've decided to have a separate `Compact` section to make it similar to all clients. `CompactConfig` has the private `serializers map[string]CompactSerializer` field and public `SetSerializers` method, similar to global/custom serializers.

**Pull request merged**